pkg/resource/deploy/source_refresh.go

```diff
@@ -96,8 +96,19 @@ func (iter *refreshSourceIterator) Next() (SourceEvent, error) {
 		// performant method at some point.
 		if iter.lastEvent != nil {
 			logging.V(7).Infof("refreshSourceIterator.Next(): waiting for previous event to retire")
+
+			var done chan struct{}
+			switch e := iter.lastEvent.(type) {
```

Comment:
nit: we could just store the `done` channel rather than the whole `SourceEvent` and thus avoid this switch.

Reply:
true, will fix.